src/librustc_mir/monomorphize/collector.rs

```diff
@@ -466,7 +466,17 @@ fn check_type_length_limit<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
                                      instance: Instance<'tcx>)
 {
     let type_length = instance.substs.types().flat_map(|ty| ty.walk()).count();
-    debug!(" => type length={}", type_length);
+    let const_length = instance.substs.consts()
+        .filter_map(|ct| {
+            if let ty::LazyConst::Evaluated(ct) = ct {
```

Comment:
It feels wrong to filter out parameters. Can you explain what's going on here?

Reply:
This was mainly intended to be a conservative extension of the type length limit check to also take account of the types in consts. This might not actually be necessary at all; it's hard to know without concrete examples of lengthy type-checking. I think `LazyConst::Unevaluated` was filtered out because getting a type from it wasn't immediately obvious.